robot_name/robot_name.rb

```diff
@@ -1,26 +1,35 @@
 class NameCollisionError < RuntimeError; end
 
 class Robot
+
+
   attr_accessor :name
 
-  @@registry
 
-  def initialize(args = {})
+  def initialize
     @@registry ||= []
-    @name_generator = args[:name_generator]
+    name_generator
+    check_registry
```

Comment:
Ideally, we should not change the logic when refactoring. I can no longer pass in a custom name generator.

(You mentioned this in the comment.)